### ISSUE TITLE
How does TFLite model and TFLite-quantized model converted?

### ISSUE THREAD

**Issue by jwj04ok** [tensorflow/tensorflow#49119]
### 1. System information
 Linux Ubuntu 16.04

### 2. Question
I downloaded two tflite models from https://www.tensorflow.org/lite/guide/hosted_models
1. https://storage.googleapis.com/download.tensorflow.org/models/mobilenet_v1_2018_02_22/mobilenet_v1_0.75_224.tgz
2.https://storage.googleapis.com/download.tensorflow.org/models/mobilenet_v1_2018_08_02/mobilenet_v1_0.75_224_quant.tgz
They are both Mobilenet_V1_0.75_224 but one if float version and one is quant version. 

When I checked the weights in first CONV (MobilenetV1/Conv2d_0/weights) node, I can notice that the first value in the float model is 0.15756385028362274 and the first value in the quantized model is 171. However, as the quantization information provided in quantized model, the float value is 0.018297843635082245 * (q - 153), which is 0.3293612003326416 for q == 171, which is not matched with float model.

May I know why is that? Does it mean that although two models are both from  Mobilenet_V1_0.75_224 but they are not converted from each other?

Also, even if two models weights are not matched, I can get nearly the same output classification when I tried several pictures. It seems that both models are correct but with different weights. Does it mean that two models are trained in different ways?

**Comment by abattery** [comment]
@daverim could you triage this issue?

**Comment by daverim** [comment]
These are indeed different models, mobilenet_v1_0.75_224_quant.tgz was trained with quantization aware training for best possible accuracy and so has a slightly different structure

**Comment by jwj04ok** [comment]
> These are indeed different models, mobilenet_v1_0.75_224_quant.tgz was trained with quantization aware training for best possible accuracy and so has a slightly different structure

Thanks, in that case, is there will be any problem if I transferred back the mobilenet_v1_0.75_224_quant to non_quant version by using the quantization info provided in the tflite file? I tried with several inputs and the middle layer outputs getting bigger and bigger and lost track in the softmax layer at the last (the power becomes 10^6 level and exp() cannot handle that). I think that's because compared with normal mobilenet_v1_0.75_224, mobilenet_v1_0.75_224_quant does not have relu6 at the end of each conv, which constrain the output min/max. However, since the mobilenet_v1_0.75_224_quant provided the quantization min/max  at each layer, I think it should ensure that whatever the input is, the output would not exceed the min/max constrain, am I correct?

Or is there another way I could do to convert the  mobilenet_v1_0.75_224_quant to its non_quant version?

Thanks!
